src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/PropertyInfoPass.php

```diff
@@ -31,17 +31,26 @@ public function process(ContainerBuilder $container)
             return;
         }
 
+        if ($container->hasDefinition('serializer.mapping.class_metadata_factory')) {
+            $definition = $container->register('property_info.serializer_extractor', 'Symfony\Component\PropertyInfo\Extractor\SerializerExtractor');
+            $definition->addArgument(new Reference('serializer.mapping.class_metadata_factory'));
+            $definition->setPublic(false);
+            $definition->addTag('property_info.list_extractor', array('priority' => -999));
+        }
```

Comment:
I understand, but when we configure it in `serializer.xml` it will only be registered when `serializer.mapping.class_metadata_factory` would be registered too.